src/main/java/org/elmlang/intellijplugin/features/intention/imports/ElmAddImportHelper.java

```diff
@@ -0,0 +1,199 @@
+            return ctors != null && ctors.isExposingAll();
+        }) ;
+
+        if (exposingAll) {
+            return typeName + "(..)";
+        }
+
+        String body;
+        body = unions.stream()
+                .filter(e -> e.getExposedUnionConstructors() != null)
+                .flatMap(e -> e.getExposedUnionConstructors().getUpperCaseIdList().stream())
+                .sorted((e1, e2) -> {
+                    String type1 = e1.getText();
+                    String type2 = e2.getText();
+                    return type1.compareTo(type2);
+                })
+                .map(e -> e.getName())
+                .collect(Collectors.joining(", "));
+
+        return body.isEmpty() ? typeName : typeName + "(" + body + ")";
+    }
+
+    private static ASTNode getInsertPosition(ElmFile sourceFile, String moduleName) {
+        Project project = sourceFile.getProject();
+        ASTNode insertPosition = null;
+        List<ElmImportClause> existingImportClauses = sourceFile.getImportClauses();
+        if (existingImportClauses.isEmpty()) {
+            // we need to create a new import section
+            PsiElement moduleDecl = sourceFile.getModuleDeclaration().orElse(null);
+
+            if (moduleDecl == null) {
+                // source file does not have an explicit module declaration
+                // so just insert at the front of the file.
+                insertPosition = sourceFile.getNode().getFirstChildNode();
+            } else {
+                // skip over comment blocks
+                PsiElement nextNonCommentSibling = PsiTreeUtil.skipSiblingsForward(moduleDecl, PsiComment.class);
+
+                // insert blanklines flanking the new import section
+                ASTNode newFreshline = ElmElementFactory.createFreshLine(project).getNode();
+                sourceFile.getNode().addChild(newFreshline, nextNonCommentSibling.getNode());
+                ASTNode newFreshline2 = ElmElementFactory.createFreshLine(project).getNode();
+                sourceFile.getNode().addChild(newFreshline2, newFreshline);
+                insertPosition = newFreshline.getTreeNext();
+            }
+        } else {
+            // find the sorted position within the existing import section
+
+            // NOTE: we *assume* that the imports are already sorted and we
+            // do not make any distinction between import groups/sections
+            // (e.g. the practice of putting core libs in the first group,
+            // 3rd party libs in a second group, and project files in the
+            // final group). In the future we will likely want to revisit
+            // this shortcut as it would very frustrating for a developer
+            // who "curates" their import list to keep fighting where the
+            // quick-fix puts its imports.
+            ElmImportClause prevImport = null;
+            for (ElmImportClause currentImport : existingImportClauses) {
```

Comment:
Can you please extract a separate function for this loop? You can also try to implement it in more functional style.